### PR TITLE
Use python SMSUsage sanitizer / munging code

### PR DESCRIPTION
**Pull request by sastels**
# Summary | Résumé

The SMSUSage data can have multiple lines per message id. This leads to double counting the cost of some notifications.

This PR changes from using a shell script to remove the PII to using a python script to remove the PII and dedup the rows.

## Related Issues | Cartes liées

* https://app.zenhub.com/workspaces/notify-planning-core-6411dfb7c95fb80014e0cab0/issues/gh/cds-snc/notification-planning-core/325

# Test instructions | Instructions pour tester la modification

- `cd scripts/sanitize_sms_usage_logs`
- `pip install -r requirements.txt` (say, in your terraform devcontainer)
- create a .env file
- run `python sanitize_sms_usage_logs.py notification-canada-ca-staging-sms-usage-logs`
- verify that the (locally saved) logs have been sanitized and deduped
- create a new bucket in staging and run
```
python sanitize_sms_usage_logs.py notification-canada-ca-staging-sms-usage-logs <mybucket> --push
```
- verify that the sanitized & deduped files have been pushed to your bucket.

Also tested in staging by running on `notification-canada-ca-staging-sms-usage-logs-san`, writing back to `notification-canada-ca-staging-sms-usage-logs-san`
```
python sanitize_sms_usage_logs.py notification-canada-ca-staging-sms-usage-logs-san notification-canada-ca-staging-sms-usage-logs-san --push
```
I then manually refreshed the dataset in QuickSight using this data and verified that the dataset was not broken.

Tested the Github action by [running the action as a PR](https://github.com/cds-snc/notification-terraform/actions/runs/9192701464/job/25282005380?pr=1335) action and then refreshing the dataset again manually.

# Release Instructions | Instructions pour le déploiement

None, though we'll have to run the script manually on the `*-san` directories to dedup the data we've already collected.

# Reviewer checklist | Liste de vérification du réviseur

- [ ] This PR does not break existing functionality.
- [ ] This PR does not violate GCNotify's privacy policies.
- [ ] This PR does not raise new security concerns. Refer to our GC Notify Risk Register document on our Google drive.
- [ ] This PR does not significantly alter performance.
- [ ] Additional required documentation resulting of these changes is covered (such as the README, setup instructions, a related ADR or the technical documentation).

> ⚠ If boxes cannot be checked off before merging the PR, they should be moved to the "Release Instructions" section with appropriate steps required to verify before release. For example, changes to celery code may require tests on staging to verify that performance has not been affected.

### CODE REVIEW

File: scripts/sanitize_sms_usage_logs/sanitize_sms_usage_logs.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    
+    if args.in_bucket is None:
+        raise ValueError("Input bucket is required")
+    in_bucket = s3.Bucket(args.in_bucket)
+
+    for key in [csv_file.key for csv_file in list(in_bucket.objects.all()) if csv_file.key.endswith(".gz") or csv_file.key.endswith(".csv")]:
+        print(f"\nProcessing {key}")
+        df = pd.read_csv(f"s3://{args.in_bucket}/{key}").sort_values(by="PublishTimeUTC")
+        df = df.drop(columns=["DestinationPhoneNumber"], errors='ignore')
+        df = df.drop_duplicates(subset=['MessageId'], keep='last')
```
Comment (jimleroyer) [inline scripts/sanitize_sms_usage_logs/sanitize_sms_usage_logs.py:31]: You tested this locally I guess? 👀

File: scripts/sanitize_sms_usage_logs/sanitize_sms_usage_logs.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        df["PriceInUSDPerFragment"] = df.PriceInUSD / df.TotalParts
```
Comment (jimleroyer) [inline scripts/sanitize_sms_usage_logs/sanitize_sms_usage_logs.py:32]: Nice!

**Review by jimleroyer — APPROVED** [review]
LGTM!

File: scripts/sanitize_sms_usage_logs/sanitize_sms_usage_logs.py (hunk 1 above)
Comment (sastels) [reply scripts/sanitize_sms_usage_logs/sanitize_sms_usage_logs.py:31]: yep!

**Pull request merged**